filament/src/MaterialParser.cpp

```diff
@@ -111,6 +112,17 @@ bool MaterialParser::operator==(MaterialParser const& rhs) const noexcept {
     if (mImpl.mManagedBuffer.size() != rhs.mImpl.mManagedBuffer.size()) {
         return false;
     }
+    std::optional<uint32_t> lhsCrc32;
+    std::optional<uint32_t> rhsCrc32;
+    if (!getMaterialCrc32(&lhsCrc32.value())) {
+        lhsCrc32 = mCrc32.load(std::memory_order_relaxed);
+    }
+    if (!rhs.getMaterialCrc32(&rhsCrc32.value())) {
+        rhsCrc32 = rhs.mCrc32.load(std::memory_order_relaxed);
+    }
+    if (lhsCrc32 && rhsCrc32 && *lhsCrc32 != *rhsCrc32) {
+        return false;
+    }
```

Comment:
this logic is pretty dense. What if you just did `computeCrc32() != rhs.computeCrc32()` ? I guess that would miss the stored crc32, but maybe that's ok?

Reply:
I pushed another commit to simplify it a bit. What do you think?